### ISSUE TITLE
Error when using `tryCatch()` with `across()`

### ISSUE THREAD

**Issue by moodymudskipper** [tidyverse/dplyr#6251]
The dot cannot be found if we use the formula notation, but is found with a standard anonymous function :

```
library(tidyverse)

mutate(band_members, across(1, ~tryCatch(stop(), error = function(e) .)))
#> Error in `mutate()`:
#> ! Problem while computing `..1 = across(1, ~tryCatch(stop(), error =
#>   function(e) .))`.
#> Caused by error in `across()`:
#> ! Problem while computing column `name`.
#> Caused by error:
#> ! object '.' not found

mutate(band_members, across(1, function(.) tryCatch(stop(), error = function(e) .)))
#> # A tibble: 3 × 2
#>   name  band   
#>   <chr> <chr>  
#> 1 Mick  Stones 
#> 2 John  Beatles
#> 3 Paul  Beatles
```

This works though: 

```
map(1, ~tryCatch(stop(), error = function(e) .))
#> [[1]]
#> [1] 1
```

**Comment by krlmlr** [comment]
It seems that `across()` might want to use `rlang::as_function()` but doesn't?

``` r
library(tidyverse)

band_members %>%
  mutate(across(1, rlang::as_function(~ tryCatch(stop(), error = function(e) .))))
#> # A tibble:      3 × 2
#> # Focus columns: name
#>   name  band   
#>   <chr> <chr>  
#> 1 Mick  Stones 
#> 2 John  Beatles
#> 3 Paul  Beatles
```

<sup>Created on 2022-05-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by krlmlr** [comment]
See also #6073 -- `across()` supports dots which might be the reason it doesn't use `rlang::as_function()` .

**Comment by hadley** [comment]
Thanks for filing this issue! Unfortunately, I think it's out of scope for this package: across is designed for simple function, not to support every possible thing you might possible put inside of it. If you need to use `tryCatch()`, I'd recommend creating a named function. Even though I'm closing this issue, I really appreciate the feedback, and hope you'll continue to contribute in the future 😄

**Comment by DavisVaughan** [comment]
It is a little weird that it works if you supply `...`

``` r
library(tidyverse)

mutate(
  band_members, 
  across(1, ~tryCatch(stop(), error = function(e) .))
)
#> Error in `mutate()`:
#> ! Problem while computing `..1 = across(1, ~tryCatch(stop(), error =
#>   function(e) .))`.
#> Caused by error in `across()`:
#> ! Problem while computing column `name`.
#> Caused by error in `value[[3L]]()`:
#> ! object '.' not found

mutate(
  band_members, 
  across(1, ~tryCatch(stop(), error = function(e) .), dot = "dot")
)
#> # A tibble: 3 × 2
#>   name  band   
#>   <chr> <chr>  
#> 1 Mick  Stones 
#> 2 John  Beatles
#> 3 Paul  Beatles
```

I have a feeling this will probably get resolved when #6073 is done, but I am not sure. I'll add a note there too

It seems to happen because `expand_across()` passes `inline = TRUE` to `across_setup()` and that is VERY CLEAR about the fact that it avoids calling `rlang::as_function()` on formulas in that case. 

https://github.com/tidyverse/dplyr/blob/6259a40b49176e3e7e8cde101376bfba41c18f53/R/across.R#L523
https://github.com/tidyverse/dplyr/blob/6259a40b49176e3e7e8cde101376bfba41c18f53/R/across.R#L355-L357
